metrics/metrics/src/main/java/io/helidon/metrics/HelidonConcurrentGauge.java

```diff
@@ -183,17 +210,31 @@ public synchronized void dec() {
         }
 
         public synchronized void updateState() {
-            long currentMinute = currentTimeMinute();
-            long diff = currentMinute - lastMinute.get();
+            if (!usingSystemClock) {
+                updateCurrentMinute();
+            }
+            long diff = currentMinute.get() - lastMinute.get();
             if (diff >= 1L) {
                 lastMax.set(currentMax.get());
                 lastMin.set(currentMin.get());
-                lastMinute.set(currentMinute);
+                lastMinute.set(currentMinute.get());
             }
         }
 
         private long currentTimeMinute() {
-            return clock.milliTime() / 1000 / 60;
+            return currentTimeMinute(clock);
+        }
+
+        private static long currentTimeMinute(Clock clock) {
+            return clock.milliTime() / 1000 / 10;
+        }
+
+        private long updateCurrentMinute() {
+            return currentMinute.getAndSet(currentTimeMinute());
```

Comment:
It seems no one consumes the value returned by this `getAndSet`. Use `set`. In this context `lazySet` is also fine.